adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java

```diff
@@ -654,14 +655,32 @@ private void closeCommandReceiverLink(final AtomicReference<MessageConsumer> mes
             final AtomicLong timerId = new AtomicLong();
 
             // create a handler for being invoked if a command was received
-            final Handler<Void> commandReceivedHandler = v -> {
+            final Handler<Message> commandReceivedHandler = commandMessage -> {
                 if (downstreamMessageSent.get()) {
                     cancelResponseTimer(timerId.get());
                     closeCommandReceiverLink(messageConsumerRef);
                     if (!ctx.response().closed()) {
                         ctx.response().end();
                     }
                 }
+                final String replyId = getReplyToIdFromCommand(tenant, deviceId, commandMessage).get();
+
+                // send answer to caller via sender link
+                final Future<CommandResponseSender> responseSender = createCommandResponseSender(tenant, deviceId, replyId);
+                responseSender.map(commandResponseSender ->
+                    commandResponseSender.sendCommandResponse(tenant, deviceId, replyId, getCorrelationIdFromMessage(commandMessage),
+                            Buffer.buffer("ok"), null, HttpURLConnection.HTTP_OK)
```

Comment:
A response without a body (null) would be more straightforward FMPOV.

Reply:
Fine for me too - will change it.